src/script/sign.cpp

```diff
@@ -73,15 +73,14 @@ static bool GetPubKey(const SigningProvider& provider, SignatureData& sigdata, c
     return false;
 }
 
-static bool CreateSig(const BaseSignatureCreator& creator, SignatureData& sigdata, const SigningProvider& provider, std::vector<unsigned char>& sig_out, const CKeyID& keyid, const CScript& scriptcode, SigVersion sigversion)
+static bool CreateSig(const BaseSignatureCreator& creator, SignatureData& sigdata, const SigningProvider& provider, std::vector<unsigned char>& sig_out, const CPubKey& pubkey, const CScript& scriptcode, SigVersion sigversion)
 {
+    CKeyID keyid = pubkey.GetID();
     const auto it = sigdata.signatures.find(keyid);
     if (it != sigdata.signatures.end()) {
         sig_out = it->second.second;
         return true;
     }
-    CPubKey pubkey;
-    GetPubKey(provider, sigdata, keyid, pubkey);
```

Comment:
Removing this is also breaking the PSBT updating test. Without it, we don't know which pubkeys are needed for signing in order to properly update a PSBT with the pubkeys.

Reply:
sorry can you point to where this breaks stuff exactly, non-obvious to me

Reply:
PSBT relies on the `GetPubKey` helper function to find out which public keys are used and to add the public key to a `SignatureData` object. The public keys are taken from the sigdata after `ProduceSignature` is done and moved into the PSBT. Without this call to `GetPubKey`, the pubkeys that were needed during signing are not added to the sigdata and thus not added to the PSBT.

Reply:
Possibly fixed.